File: MyApp/app/src/main/java/com/example/myapp/data/model/Customer.kt
package com.example.myapp.data.model

import com.google.gson.annotations.SerializedName

data class CustomerResponse(
    val code: Int,
    val description: String,
    val data: CustomerData
)

data class CustomerData(
    val customers: List<Customer>,
    val page: Page
)

data class Customer(
    val id: String,
    val name: String,
    val email: String,
    val address: String? = null,
    @SerializedName("created_at") val createdAt: String? = null,
    @SerializedName("updated_at") val updatedAt: String? = null,
    val phone: String? = null,
    @SerializedName("orders_count") val ordersCount: String? = null
)

File: my_flutter_order_management_system_app/lib/data/models/customer.dart
import 'package:my_flutter_order_management_system_app/data/models/order.dart';
import 'package:my_flutter_order_management_system_app/data/models/product.dart';

class CustomerResponse {
  final int code;
  final String description;
  final CustomerData data;

  CustomerResponse({required this.code, required this.description, required this.data});

  factory CustomerResponse.fromJson(Map<String, dynamic> json) {
    return CustomerResponse(
      code: json['code'],
      description: json['description'],
      data: CustomerData.fromJson(json['data']),
    );
  }
}

class CustomerUpdateResponse {
  final int code;
  final String description;
  final Customer data;

  CustomerUpdateResponse({required this.code, required this.description, required this.data});

  factory CustomerUpdateResponse.fromJson(Map<String, dynamic> json) {
    return CustomerUpdateResponse(
      code: json['code'],
      description: json['description'],
      data: Customer.fromJson(json['data']),
    );
  }
}

class CustomerData {
  final List<Customer> customers;
  final Page page;

  CustomerData({required this.customers, required this.page});

  factory CustomerData.fromJson(Map<String, dynamic> json) {
    return CustomerData(
      customers: (json['customers'] as List).map((e) => Customer.fromJson(e)).toList(),
      page: Page.fromJson(json['page']),
    );
  }
}

